test/framework/cluster_proxy.go

```diff
@@ -247,20 +266,59 @@ func (p *clusterProxy) GetCache(ctx context.Context) cache.Cache {
 	return p.cache
 }
 
-// Apply wraps `kubectl apply ...` and prints the output so we can see what gets applied to the cluster.
-func (p *clusterProxy) Apply(ctx context.Context, resources []byte, args ...string) error {
-	Expect(ctx).NotTo(BeNil(), "ctx is required for Apply")
-	Expect(resources).NotTo(BeNil(), "resources is required for Apply")
-
-	if err := testexec.KubectlApply(ctx, p.kubeconfigPath, resources, args...); err != nil {
-		var exitErr *exec.ExitError
-		if errors.As(err, &exitErr) {
-			return pkgerrors.New(fmt.Sprintf("%s: stderr: %s", err.Error(), exitErr.Stderr))
+// Creates or updates objects using the clusterProxy client.
+func (p *clusterProxy) CreateOrUpdate(ctx context.Context, resources []byte, opts ...CreateOrUpdateOption) error {
+	Expect(ctx).NotTo(BeNil(), "ctx is required for Create")
+	Expect(resources).NotTo(BeNil(), "resources is required for Create")
+	labelSelector := labels.Everything()
+	config := &CreateOrUpdateConfig{}
+	for _, opt := range opts {
+		opt(config)
+	}
+	if config.LabelSelector != "" {
+		ls, err := metav1.ParseToLabelSelector(config.LabelSelector)
+		if err != nil {
+			return fmt.Errorf("could not parse selector: %+w", err)
+		}
+		labelSelector, err = metav1.LabelSelectorAsSelector(ls)
+		if err != nil {
+			return fmt.Errorf("could not convert metav1.LabelSelector to labels.Selector: %+w", err)
 		}
+	}
```

Comment:
```suggestion
	if config.labelSelector != nil {
		labelSelector = config.labelSelector
	}
```

With the above changes :-)